salt/minion_auth/README.md

```diff
@@ -0,0 +1,47 @@
+# Foreman Salt Minion Authentication
+
+Currently, there are two possibilites to authenticate a newly deployed Minion automatically:
+1. Use the /etc/salt/autosign.conf file which stores the hostnames of acceptable hosts.
+2. Use Salt Autosign Grains for a more secure way which relies on a shared secret key.
+
+This README, handles the second option and how to configure it
+
+## Setup
+Add the content of 'master.snippet' to '/etc/salt/master' which configures the Grains key file on the Master and a Reactor. The Grains file holds the acceptable keys and will be written by the Smart Proxy when a new Minion is deployed. The Reactor initiates an interaction with Foreman Salt if a new Minion was authenticated successfully.
+In case there is already a Reactor configured, you need to adapt it using the options mentioned in 'master.snippet'. The directories given in 'master.snippet' are the default ones. In case you want your files in a different place, you have to change the paths accordingly.
+
+If '/srv/salt' is configured as 'file_roots' in your '/etc/salt/master' config, setup the necessary Salt Runners:
+
+```
+/srv/salt/_runners/foreman_file.py
+/srv/salt/_runners/foreman_https.py
+```
+
+Check if the Reactor ('foreman_minion_auth.sls') is at the appropriated location:
+
+```
+/var/lib/foreman-proxy/salt/reactors/foreman_minion_auth.sls
+```
+
+After changing the Salt Master run:
+
+```
+systemctl restart salt-master
+```
+
+After checking the Reactor and Runners, run the following:
+
+```
+salt-run saltutil.sync_all
+```
+
+## Procedure
+
+1. A new host, configured as Salt Minion, is deployed with Foreman Salt.
+2. Foreman Salt generates a unique key for that Minion and distributes it via the Provisioning Template to the host and via an API call to the Smart Proxy.
+3. The Smart Proxy makes the key available for the Salt Autosign Grains procedure by adding it to the previously defined file (by default: /var/lib/foreman-proxy/salt/grains/autosign_key).
+4. The Salt Master accepts the Minion depending on the key and the corresponding auth reactor is triggered on the Salt Master.
+5. The Salt Master initiates an API call to Foreman Salt which marks the corresponding host status as	_authenticated_.
+6. Foreman Salt triggers an API call to Smart Proxy Salt which deletes the key from the acceptable keys list (since the Minion was authenticated already).
```

Comment:
another steop: Foreman salt removes the already used salt autosign grain key for the authenticated salt minion.

Reply:
Step 6 describes this. I could add the following to make it more clear:
```
6. Foreman Salt triggers an API call to Smart Proxy Salt which deletes the key from the acceptable keys list of the Salt Master (since the Minion was authenticated already)
```